gwdetchar/io/html.py

```diff
@@ -441,6 +441,71 @@ def write_arguments(content, start, end, flag=None, section='Parameters',
     return page()
 
 
+def table(headers, data, caption=None, separator='', id=None, **class_):
+    """Write a <table> with one row of headers and many rows of data
+
+    Parameters
+    ----------
+    headers : `list`
+        list of column header names
+
+    data : `list` of `lists`
+        list of column data lists, for ``m`` rows and ``n`` columns, this
+        should have dimensions ``m x n``
+
+    caption : `str`, optional
+        content for this table's `<caption>`
+
+    **class_
+        class attribute declarations for each tag used in the table,
+        any of `table`, `thead`, `tbody`, `tr`, `th`, `td`, `caption`
+
+    Returns
+    -------
+    table : `markup.page`
+        a formatted HTML page object containing the `<table>`
+    """
+    class_.setdefault('table',
+                      'table table-hover table-condensed table-responsive')
+    # unwrap class declarations (so we don't get empty class attributes)
+    kwargs = {}
+    for tag in ['table', 'thead', 'tbody', 'tr', 'th', 'td', 'caption']:
+        try:
+            kwargs[tag] = {'class_': class_.pop(tag)}
+        except KeyError:
+            kwargs[tag] = {}
+    # create table and add caption
+    page = markup.page(separator=separator)
+    if id is not None:
+        kwargs['table']['id_'] = id
+    page.table(**kwargs['table'])
+    if caption:
+        page.caption(caption, **kwargs['caption'])
+    # write headers
+    page.thead(**kwargs['thead'])
+    page.tr(**kwargs['tr'])
+    for th in headers:
+        page.th(th, scope='col', **kwargs['th'])
+    page.tr.close()
+    page.thead.close()
+    # write body
+    page.tbody(**kwargs['tbody'])
+    for row in data:
+        page.tr(**kwargs['tr'])
+        for td in row:
+            page.td(td, **kwargs['td'])
+        page.tr.close()
+    page.tbody.close()
+    page.table.close()
+    # add export button
+    if id is not None:
```

Comment:
```suggestion
    if id:
```

(since you don't want to set an empty string as the `id` either